k8sutil/kube_config.py

```diff
@@ -0,0 +1,160 @@
+# Copyright 2016 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import atexit
+import base64
+import os
+import tempfile
+import urllib3
+import yaml
+
+from oauth2client.client import GoogleCredentials
+
+from k8sclient import configuration
+
+_temp_files = []
+
+
+def _cleanup_temp_files():
+    for f in _temp_files:
+        os.remove(f)
+
+
+def _create_temp_file_with_content(content):
+    if len(_temp_files) == 0:
+        atexit.register(_cleanup_temp_files)
+    _, name = tempfile.mkstemp()
+    _temp_files.append(name)
+    if isinstance(content, str):
+        content=content.encode('utf8')
+    with open(name, 'wb') as fd:
+        fd.write(base64.decodestring(content))
+    return name
+
+
+def _find_object_with_name(o, name, o_name):
```

Comment:
This whole function was obsolete. sorry for that. removed. (name was the value of attribute 'name' of the object, and o_name was a debug friendly name of the object such as 'user' or 'cluster'. the logic moved to _node.get_with_name)